### PR TITLE
feat/structured_data: support deletion

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/structured_data.rs [hunk 1]
```diff
@@ -96,6 +96,32 @@ impl StructuredData {
         DataIdentifier::Structured(self.name, self.type_tag)
     }
 
+    /// Delete self
+    pub fn delete_self_against_successor(&mut self,
```
Comment (afck) [inline src/structured_data.rs]: Maybe `delete_if_valid_successor` would make more sense as a name.

File: src/structured_data.rs [hunk 2]
```diff
@@ -96,6 +96,32 @@ impl StructuredData {
...
+                                         other: &StructuredData)
+                                         -> Result<(), RoutingError> {
+        if !(other.data.is_empty() && other.current_owner_keys.is_empty()) {
+            return Err(RoutingError::UnknownMessageType);
+        }
+        match self.validate_self_against_successor(other) {
```
Comment (afck) [inline src/structured_data.rs]: Maybe just `try!(self.validate...)`. Then no `match` is needed.

**Comment by ustulation** [comment]
Merged into https://github.com/maidsafe/routing/pull/1149

**Pull request closed without merging**